jsapp/js/components/modalForms/translationSettings.es6

```diff
@@ -0,0 +1,324 @@
+    }
+  }
+  nameChange (e) {
+    this.setState({languageName: e.target.value});
+  }
+  codeChange (e) {
+    this.setState({languageCode: e.target.value});
+  }
+  render () {
+    let fieldsNotEmpty = this.state.languageName.length > 0 && this.state.languageCode.length > 0;
+    var btnClasses = classNames('mdl-button','mdl-js-button', 'mdl-button--raised', fieldsNotEmpty ? 'mdl-button--colored' : 'mdl-button--disabled');
+
+    return (
+      <bem.FormView__cell m='add-language-fields'>
+        <bem.FormView__cell>
+          <bem.FormModal__item>
+            <label>{t('Language name')}</label>
+            <input type='text' value={this.state.languageName} onChange={this.nameChange} />
+          </bem.FormModal__item>
+        </bem.FormView__cell>
+        <bem.FormView__cell m='lang-code'>
+          <bem.FormModal__item>
+            <label>{t('Language code')}</label>
+            <input type='text' value={this.state.languageCode} onChange={this.codeChange} />
+          </bem.FormModal__item>
+        </bem.FormView__cell>
+        <bem.FormView__cell>
+          <button className={btnClasses} onClick={this.submit}>
+            {this.props.langIndex !== undefined ? t('Update') : t('Add')}
+          </button>
+        </bem.FormView__cell>
+      </bem.FormView__cell>
+      );
+  }
+};
+
+export class TranslationSettings extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      translations: props.asset.content.translations || [],
+      showAddLanguageForm: false,
+      renameLanguageIndex: -1
+    }
+    autoBind(this);
+  }
+  componentDidMount () {
+    this.listenTo(stores.asset, this.assetStoreChange);
+  }
+  assetStoreChange(asset) {
+    let uid = this.props.asset.uid;
+
+    this.setState({
+      translations: asset[uid].content.translations,
+      showAddLanguageForm: false,
+      renameLanguageIndex: -1
+    })
+
+    stores.pageState.showModal({
+      type: 'form-languages',
+      asset: asset[uid]
+    });
+  }
+  showAddLanguageForm() {
+    this.setState({
+      showAddLanguageForm: true
+    })
+  }
+  hideAddLanguageForm() {
+    this.setState({
+      showAddLanguageForm: false
+    })
+  }
+  toggleRenameLanguageForm(e) {
+    let index = parseInt($(e.target).closest('[data-index]').get(0).getAttribute('data-index'));
+    if (this.state.renameLanguageIndex === index) {
+      this.setState({
+        renameLanguageIndex: -1
+      });
+    } else {
+      this.setState({
+        renameLanguageIndex: index
+      });
+    }
+  }
+  launchTranslationTableModal(e) {
+    let index = parseInt($(e.target).closest('[data-index]').get(0).getAttribute('data-index')),
+        asset = this.props.asset;
+    stores.pageState.hideModal();
+    window.setTimeout(function(){
+      stores.pageState.showModal({
+        type: 'form-translation-table',
+        asset: asset,
+        langIndex: index
+      });
+    }, 300);
```

Comment:
Is it possible to wait for some event/action rather than random `ms` value? I'm ok with that if too hard to implement